gh-connector/src/main/java/de/tum/gh_connector/service/GHConnectorService.java

```diff
@@ -0,0 +1,127 @@
+                .build();
+    }
+
+    private void crawlWorkflows(String searchPath, String basePath, List<WorkflowFile> resulList) {
+        List<ContentResponseItem> contents = ghRestClient.getFolderContent(basePath + searchPath);
+
+        for (ContentResponseItem item : contents) {
+            try {
+
+                if (item.getType().equals("dir")) {
+                    crawlWorkflows(item.getPath(), basePath, resulList);
+                }
+
+                if (item.getType().equals("file")
+                        && (item.getName().endsWith(".yml") || item.getName().endsWith(".yaml"))) {
+                    ContentResponseItem contentItem = ghRestClient.getFileContent(basePath + item.getPath());
+                    resulList.add(WorkflowFile.fromContentResponseItem(contentItem));
+                }
+            } catch (Exception e) {
+                // todo: rate exeeded
+                e.printStackTrace();
+
+            }
+        }
+    }
+
+    private void assertWorkflowDirAccess(String contentPath) throws IllegalArgumentException {
+        try {
+            ghRestClient.getFolderContent(contentPath);
+        } catch (FeignException.NotFound e) {
+            throw new IllegalArgumentException("The specified Repository doesn't exist or you are not authorized to access it");
+        }
+
+        String workflowsPath = contentPath + "/.github/workflows";
+
+        try {
+            ghRestClient.getFolderContent(workflowsPath);
+        } catch (FeignException.NotFound e) {
+            throw new IllegalArgumentException("The specified Repository doesn't have a workflow directory");
+        }
+    }
+
+    private String constructGHApiContentPath(String repoUri) throws IllegalArgumentException {
+        URI uri = URI.create(repoUri);
+
+        if (!"https".equals(uri.getScheme())) {
+            throw new IllegalArgumentException("The provided URL is not HTTPS: " + uri);
+        }
+
+        if (!"github.com".equals(uri.getHost())) {
+            throw new IllegalArgumentException("URL does not point to github.com");
+        }
+
+        String path = uri.getPath();
+        if (path == null || path.isEmpty()) {
+            throw new IllegalArgumentException("URL has no specified path.");
+        }
+
+        String[] pathParts = path.split("/");
+        if (pathParts.length < 3 || "".equals(pathParts[1]) || "".equals(pathParts[2])) {
+            throw new IllegalArgumentException("URL Path is not long enough - The repository is not clear.");
+        }
+
+        if (!"".equals(pathParts[0])) {
+            throw new IllegalArgumentException("This error should never occur");
+        }
+
+        return "/repos/" + pathParts[1] + "/" + pathParts[2] + "/contents/";
+    }
+
+    private GHConnectorResponse constructError(String message) {
+        return GHConnectorResponse.builder()
+                .status("400")
+                .message(message)
+                .build();
+    }
```

Comment:
I don't really like having a http status in the payload. Why not use the actual http status code of the response?